tests aren't all passing in master branch

Subject: test is failing with python 3.5

### Problem
Running tests after cloning master branch results in a failed unit test.

#### Procedure to reproduce the problem
I followed the procedure to setup dev environment like detailed in https://github.com/sphinx-doc/sphinx/blob/master/CONTRIBUTING.rst, I used the following command `tox -e py35`

#### Error logs / results
```
...
=================================== FAILURES ===================================
______________________________ test_partialmethod ______________________________

    @pytest.mark.skipif(sys.version_info < (3, 4),
                        reason='functools.partialmethod is available on py34 or above')
    @pytest.mark.usefixtures('setup_test')
    def test_partialmethod():
        def call_autodoc(objtype, name):
            inst = app.registry.documenters[objtype](directive, name)
            inst.generate()
            result = list(directive.result)
            del directive.result[:]
            return result
    
        options.inherited_members = True
        options.undoc_members = True
        expected = [
            '',
            '.. py:class:: Cell',
            '   :module: target.partialmethod',
            '',
            '   An example for partialmethod.',
            '   ',
            '   refs: https://docs.python.jp/3/library/functools.html#functools.partialmethod',
            '   ',
            '   ',
            '   .. py:method:: Cell.set_alive() -> None',
            '      :module: target.partialmethod',
            '   ',
            '      Make a cell alive.',
            '      ',
            '   ',
            '   .. py:method:: Cell.set_dead() -> None',
            '      :module: target.partialmethod',
            '   ',
            '      Make a cell dead.',
            '      ',
            '   ',
            '   .. py:method:: Cell.set_state(state)',
            '      :module: target.partialmethod',
            '   ',
            '      Update state of cell to *state*.',
            '      ',
        ]
        if sys.version_info < (3, 5):
            expected = '\n'.join(expected).replace(' -> None', '').split('\n')
    
>       assert call_autodoc('class', 'target.partialmethod.Cell') == expected
E       AssertionError: assert ['', '.. py:c...', '   ', ...] == ['', '.. py:cl...', '   ', ...]
E         At index 9 diff: '   .. py:method:: Cell.set_alive()' != '   .. py:method:: Cell.set_alive() -> None'
E         Use -v to get the full diff

tests/test_autodoc.py:956: AssertionError

```

#### Expected results
Not failing the test.

### Environment info
- OS: Ubuntu 17.04
- Python version: python 3.5.3
- Sphinx version: master branch

--- a/tests/test_autodoc.py
+++ b/tests/test_autodoc.py
@@ -950,7 +950,7 @@ def test_partialmethod():
         '      Update state of cell to *state*.',
         '      ',
     ]
-    if sys.version_info < (3, 5):
+    if sys.version_info < (3, 5, 4):
         expected = '\n'.join(expected).replace(' -> None', '').split('\n')
 
     assert call_autodoc('class', 'target.partialmethod.Cell') == expected

